market_approval_server/backup_config.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Backup market approval server configuration."""
+
+import argparse
+import json
+import time
+from typing import Any, Dict, Optional
+
+import requests
+
+
+MAX_RETRIES = 3
+RETRY_DELAY_SECONDS = 10
+
+def _fetch_data(endpoint_url: str) -> Optional[Dict[str, Any]]:
+    for retry_count in range(MAX_RETRIES):
+        response = requests.get(endpoint_url)
+        if response.status_code == 200:
+            return response.json()
+        print(
+            f"Failed to fetch data from {endpoint_url}. Status code: {response.status_code}"
+        )
+        if retry_count < MAX_RETRIES - 1:
+            print(f"Retrying in {RETRY_DELAY_SECONDS} seconds...")
+            time.sleep(RETRY_DELAY_SECONDS)
+    return None
+
+
+def _process_endpoints(url: str) -> Optional[Dict[str, Any]]:
+    approved_markets = _fetch_data(f"{url}/approved_markets")
+    rejected_markets = _fetch_data(f"{url}/rejected_markets")
+    proposed_markets = _fetch_data(f"{url}/proposed_markets")
+    processed_markets = _fetch_data(f"{url}/processed_markets")
+
+    if approved_markets and rejected_markets and proposed_markets and processed_markets:
+        _backup_data: Dict[str, Any] = {
+            "api_keys": {
+                "454d31ff03590ff36836e991d3287b23146a7a84c79d082732b56268fe472823": "default_user"
+            },
+            "approved_markets": approved_markets.get("approved_markets", {}),
+            "processed_markets": processed_markets.get("processed_markets", {}),
+            "proposed_markets": proposed_markets.get("proposed_markets", {}),
+            "rejected_markets": rejected_markets.get("rejected_markets", {}),
+        }
+        return _backup_data
+
+    return None
```

Comment:
Could try something like this to return earlier:

```suggestion
    _backup_data = {"api_keys": {                "454d31ff03590ff36836e991d3287b23146a7a84c79d082732b56268fe472823": "default_user"}}
    for market_name in market_names:
        res = _fetch_data(f"{url}/market_name")
        if res is None:
            return None
        _backup_data[market_name] = res.get(market_name, {})
    return _backup_data
```

Reply:
fixed